classes/patreon_frontend.php

```diff
@@ -1073,5 +1074,41 @@ public static function LoginButtonShortcode( $args ) {
 		}
 		
 	}
+	public static function show_patreon_avatar( $avatar, $id_or_email, $size, $default, $alt ) {
```

Comment:
some default values for size & alt?

Reply:
Since show_patreon_avatar is hooked to WordPress' get_avatar filter and receives these values from there, any defaults would come from WP provided to the function via the filter. These are determined and modified by the WP functions (or plugin functions) which apply or hook to this filter. This can potentially make the following logic fail in certain setups. When hooking to filters, its better to leave all variables that are acquired from the filters to whatever value filter may or may not provide. The following example for get_avatar filter may provide insight:

https://codex.wordpress.org/Plugin_API/Filter_Reference/get_avatar

Reply:
Fair enough. Thank you for taking the time to explain